include/zephyr/net/tls_credentials.h

```diff
@@ -61,11 +64,10 @@ enum tls_credential_type {
  * Secure tag can be used to reference credential after it was registered
  * in the system.
  *
- * @note Some TLS credentials come in pairs:
+ * @note Some TLS credentials come in pairs, and must be assigned the same
```

Comment:
I will need to look more at the implementation, as it is unclear how it actually is working when they are assigned the same tag.

Reply:
It's essential for PSK/PSK_ID, the TLS socket layer expects that it finds PSK_ID under the same tag as PSK once it encounters it. For cert/priviate_key it's not a must, but it's still logically sane to keep them together under the same tag.